sdk/appconfiguration/azure-data-appconfiguration/swagger/src/main/java/AppConfigCustomization.java

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import com.azure.autorest.customization.ClassCustomization;
+import com.azure.autorest.customization.Customization;
+import com.azure.autorest.customization.LibraryCustomization;
+import com.azure.autorest.customization.MethodCustomization;
+import com.azure.autorest.customization.PackageCustomization;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.stmt.BlockStmt;
+import org.slf4j.Logger;
+
+/**
+ * This class contains the customization code to customize the AutoRest generated code for App Configuration.
+ */
+public class AppConfigCustomization extends Customization {
+
+    @Override
+    public void customize(LibraryCustomization customization, Logger logger) {
+        PackageCustomization models = customization.getPackage("com.azure.data.appconfiguration.models");
+
+        customizeKeyValueFilter(models.getClass("KeyValueFilter"));
+
+        customizeKeyValueFields(models.getClass("KeyValueFields"));
+    }
+
+    private void customizeKeyValueFilter(ClassCustomization classCustomization) {
+        classCustomization.getMethod("setLabel")
+            .getJavadoc()
+            .setDescription("Set the label property: Filters {@link ConfigurationSetting} by their label field.")
+            .setReturn("the {@link KeyValueFilter} object itself.");
+        classCustomization.getMethod("getKey")
+            .getJavadoc()
+            .setDescription("Get the key property: Filters {@link ConfigurationSetting} by their key field.");
+        classCustomization.getMethod("getLabel")
+            .getJavadoc()
+            .setDescription("Get the label property: Filters {@link ConfigurationSetting} by their label field.");
+
+        classCustomization = classCustomization.customizeAst(
+            ast -> ast.getClassByName("KeyValueFilter")
+                       .get()
+                       .getDefaultConstructor().get()
+                       .remove())
+                                 .removeMethod("setKey")
+                                 .rename("SnapshotSettingFilter");
```

Comment:
KeyValueFilter is the swagger defined name, "KeyValue" is `ConfigurationSetting` in the public API. 
Currently, the proposed name for the filter in all language is `SnapshotSettingFilter` but it could change to another name in the future. I would keep a note of it.